pkgs/servers/http/nginx/generic.nix

```diff
@@ -17,6 +17,12 @@ stdenv.mkDerivation {
     inherit sha256;
   };
 
+  patches = [
+    (fetchpatch {
```

Comment:
This could actually be `fetchurl`, but I'm not sure if I should use here or leave `fetchpatch` for consistency.

Reply:
I think it would be better to add this patch to nixpkgs repo. The general preference (as I understand it) is to use `fetchpatch` for patches from upstream, other distros or other authors, while patches written specifically for nixpkgs should be added to our repo.

Reply:
Maybe. @grahamc, what do you think?